### PR TITLE
[0.68] Update OnDropViewInstance API Doc

### PR DESCRIPTION
**Pull request by chiaramooney**
### Type of Change
_Erase all that don't apply._
- Documentation

### Why
Backport #9627 to 0.68

### What
Add API documentation for new interface. Adjust method description.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/9641)

### CODE REVIEW

**Review by TatianaKapos — APPROVED** [review]
Lgtm! :)

File: vnext/Microsoft.ReactNative/IViewManager.idl [hunk 1]
```diff
@@ -88,10 +88,18 @@ namespace Microsoft.ReactNative
...
+    "removed from their shadow trees. "
+  )
   interface IViewManagerWithDropViewInstance {
     DOC_STRING(
-        "Enables the view manager to track when the view is "
-        "being removed from the visual tree. "
+        "Invoked when React has removed the view "
+        "from its shadow tree. Note this method call is distinct from "
+        "the XAML Unloaded event, which is asynchronously triggered "
+        "from when the removal of the view from the UI tree happens. "
```
Comment (jonthysell) [inline vnext/Microsoft.ReactNative/IViewManager.idl:102]:
```suggestion
        "when the view is removed from the UI tree."
```

File: vnext/Microsoft.ReactNative/IViewManager.idl [hunk 2]
```diff
@@ -88,10 +88,18 @@ namespace Microsoft.ReactNative
     )
     void OnPointerEvent(Object view, ReactPointerEventArgs args);
   };
+
+  [webhosthidden]
+  DOC_STRING(
+    "Enables view managers to track when views are"
+    "removed from their shadow trees. "
```
Comment (jonthysell) [inline vnext/Microsoft.ReactNative/IViewManager.idl:95]: Is their more than one shadow tree?

```suggestion
    "removed from the shadow tree."
```

**Review by jonthysell — APPROVED** [review]

File: vnext/Microsoft.ReactNative/IViewManager.idl (hunk 1 above)
Comment (asklar) [reply vnext/Microsoft.ReactNative/IViewManager.idl]: @chiaramooney since this was a backport, can you please make sure the text is updated in main too?

**Pull request merged**